### PR TITLE
Fixes server log hardcoded to debug level

### LINKED ISSUE #3786: sbt server log is set to debug level
(See the guidelines for contributing, linked above)

## steps

1. Open sbt/sbt using 1.1.0-RC1.
2. Open it using VS Code.

## problem

Lots of debug level logs.

```
[debug] Unhandled notification received: textDocument/didOpen
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/src/main/scala/sbt/internal/util/IDSet.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/src/main/scala/sbt/internal/util/Util.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/src/main/scala/sbt/internal/util/Signal.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/src/main/scala/sbt/internal/util/HListFormats.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/src/main/scala/sbt/internal/util/HList.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/src/main/scala/sbt/internal/util/Classes.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/src/main/scala/sbt/internal/util/Settings.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/src/main/scala/sbt/util/Show.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/target/scala-2.12/src_managed/main/sbt/internal/util/ScalaKeywords.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/src/main/scala/sbt/internal/util/KList.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/src/main/scala/sbt/internal/util/AList.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/src/main/scala/sbt/internal/util/INode.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/src/main/scala/sbt/internal/util/Dag.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/src/main/scala/sbt/util/OptJsonWriter.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/src/main/scala/sbt/internal/util/Attributes.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/src/main/scala/sbt/internal/util/PMap.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/src/main/scala/sbt/internal/util/Types.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/src/main/scala/sbt/internal/util/TypeFunctions.scala","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/testing/agent/src/main/java/sbt/ForkTags.java","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/testing/agent/src/main/java/sbt/ForkMain.java","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/testing/agent/src/main/java/sbt/FrameworkWrapper.java","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, textDocument/publishDiagnostics, {"uri":"file:/Users/eed3si9n/work/sbt-modules/sbt/testing/agent/src/main/java/sbt/ForkConfiguration.java","diagnostics":[]})
[debug] langNotify: JsonRpcNotificationMessage(2.0, window/logMessage, {"type":3,"message":"Compiling 18 Scala sources to /Users/eed3si9n/work/sbt-modules/sbt/internal/util-collection/target/scala-2.12/classes ..."})
```

## expectation

No debug level logs.

## notes

https://github.com/sbt/sbt/blob/v1.1.0-RC1/main/src/main/scala/sbt/internal/CommandExchange.scala#L109

```scala
    val serverLogLevel: Level.Value = Level.Debug
```

### PR DESCRIPTION
**Pull request by eed3si9n**
Fixes #3786

To configure the log level of the server, this introduces a new task key named `serverLog`. The idea is to set this using `Global / serverLog / logLevel`. It will also check the global log level, and if all else fails, fallback to Warn.

```diff
+    lazy val level: Level.Value = (s get serverLogLevel)
+      .orElse(s get logLevel)
+      .getOrElse(Level.Warn)
```

### CODE REVIEW

File: main/src/main/scala/sbt/internal/CommandExchange.scala [hunk 1]
```diff
@@ -106,15 +113,19 @@ private[sbt] final class CommandExchange {
       case Some(x) => x
       case None    => ConnectionType.Tcp
     }
-    val serverLogLevel: Level.Value = Level.Debug
+    lazy val level: Level.Value = (s get serverLogLevel) orElse (s get logLevel) match {
+      case Some(x) => x
+      case None    => Level.Warn
+    }
```
Comment (dwijnand) [inline main/src/main/scala/sbt/internal/CommandExchange.scala]: `getOrElse Level.Warn`

Comment (eed3si9n) [reply main/src/main/scala/sbt/internal/CommandExchange.scala]: done

File: main/src/main/scala/sbt/internal/CommandExchange.scala [hunk 2]
```diff
@@ -106,15 +113,18 @@ private[sbt] final class CommandExchange {
       case Some(x) => x
       case None    => ConnectionType.Tcp
     }
-    val serverLogLevel: Level.Value = Level.Debug
+    lazy val level: Level.Value = (s get serverLogLevel)
```
Comment (dwijnand) [inline main/src/main/scala/sbt/internal/CommandExchange.scala:116]: shouldn't this be `s get (logLevel in serverLogLevel)`?

Comment (eed3si9n) [reply main/src/main/scala/sbt/internal/CommandExchange.scala:116]: As discussed

```scala
.setCond(serverLogLevel, srvLogLevel)
```

in Project.scala where srvLogLevel does the lookup.

**Pull request merged**